### PR TITLE
v2: re-write using `gherkin@4` and `pull-stream`

### PR DESCRIPTION
**Pull request by ahdinosaur**
hey @yoshuawuyts, so i re-wrote `cuke-tap` with the latest.
- compile to pickles data structure instead of using gherkin document
- receive glob string to feature files instead of array of paths
- change how steps are defined so there are no globals. each step is
  an array of [regex, fn], many steps are an array of these and must be
  exported and combined manually.

wonder what you think, thanks. :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,13 +19,13 @@ __test/index.js__
 const cuke = require('cuke-tap')
 const path = require('path')
 
-const features = [ path.join(__dirname, '/feature.feature') ]
-const steps = [ require('./steps') ]
+const features = path.join(__dirname, 'example.feature')
+const steps = require('./steps')
 
-cuke(features, steps)
+cuke({ steps, features })
```
Comment (yoshuawuyts) [inline README.md]: no destructures please :grin:

Comment (ahdinosaur) [inline README.md]: this isn't destructuring, it's "short-hand properties" (in Node 4). still nay?

Comment (yoshuawuyts) [inline README.md]: haha yeah rather not; just es2020 :sparkles:

**Comment by yoshuawuyts** [comment]
> compile to pickles data structure

hahaha, that's an actual thing? I love it

Everything's looking :100:, super excited to try this out! :D

File: README.md (hunk 1 above)
Comment (ahdinosaur) [inline README.md]: sweet, done. had one of those in the code as well.

**Comment by yoshuawuyts** [comment]
Published as `v2.0.0` :tada:

**Pull request merged**